documents/anchor_task.go

```diff
@@ -116,16 +120,24 @@ func (d *documentAnchorTask) RunTask() (res interface{}, err error) {
 }
 
 // InitDocumentAnchorTask enqueues a new document anchor task and returns the txID.
-func InitDocumentAnchorTask(tq queue.TaskQueuer, txService transactions.Service, accountID identity.CentID, modelID []byte) (uuid.UUID, error) {
-	tx, err := txService.CreateTransaction(accountID, documentAnchorTaskName)
+// TODO [TXManager] migrate this to use TxManager
+func InitDocumentAnchorTask(tq queue.TaskQueuer, txService transactions.Manager, accountID identity.CentID, modelID []byte, txID uuid.UUID) (uuid.UUID, error) {
+	var tx *transactions.Transaction
+	var err error
+	if txID != uuid.Nil {
```

Comment:
Will it never be the case that txID here is `nil`? or uuid.Nil is an alias as well of go nil?

Reply:
As explained offline this can't be nil